### ISSUE TITLE
gcc9 formula seems to believe glibc is installed, though it is not.

### ISSUE THREAD

**Issue by dvhwgumby** [Homebrew/homebrew-core#42761]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X ] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [ X] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X ] ran `brew update` and can still reproduce the problem?
- [X ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

```
$ brew gist-logs gcc
Error: No logs.
```
- [X ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?
```
$ brew config && brew doctor
HOMEBREW_VERSION: 2.1.9-7-gde72a23
ORIGIN: https://github.com/Homebrew/brew
HEAD: de72a2310930c3f0903292770b433f197a8d8a6c
Last commit: 23 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: f21a2271ff6f5b88e56e12f09e1c65ad7aef0201
Core tap last commit: 5 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
HOMEBREW_VISUAL: /Applications/Emacs.app/Contents/MacOS/bin/emacsclient -a /usr/bin/emacs 
CPU: quad-core 64-bit kabylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1001
Git: 2.22.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 11.0.2, 1.8.0_151
macOS: 10.14.6-x86_64
CLT: 10.3.0.0.1.1562985497
Xcode: 10.3
CLT headers: 10.3.0.0.1.1562985497
XQuartz: 2.7.11 => /opt/X11
Your system is ready to brew.
```
To help us debug your issue please explain:

- What you were trying to do (and why)
I'm trying to compile icu4c with gcc9

- What happened (include command output)
gcc is configured as if glibc is installed, though it is not.  Specifically, I get:
```
In file included from /usr/local/Cellar/gcc/9.1.0/include/c++/9.1.0/system_error:39,
                 from /usr/local/Cellar/gcc/9.1.0/include/c++/9.1.0/bits/std_mutex.h:39,
                 from /usr/local/Cellar/gcc/9.1.0/include/c++/9.1.0/condition_variable:39,
                 from ../../../3P-src/unicode/source/common/umutex.h:24,
                 from ../../../3P-src/unicode/source/common/putil.cpp:68:
/usr/local/Cellar/gcc/9.1.0/include/c++/9.1.0/x86_64-apple-darwin18/bits/error_constants.h:135:24: error: 'EOWNERDEAD' was not declared in this scope
  135 |       owner_dead =     EOWNERDEAD,
      |                        ^~~~~~~~~~
/usr/local/Cellar/gcc/9.1.0/include/c++/9.1.0/x86_64-apple-darwin18/bits/error_constants.h:151:34: error: 'ENOTRECOVERABLE' was not declared in this scope
  151 |       state_not_recoverable =    ENOTRECOVERABLE,
      |                                  ^~~~~~~~~~~~~~~
```
because  _GLIBCXX_HAVE_EOWNERDEAD and _GLIBCXX_HAVE_ENOTRECOVERABLE  are set.

- What you expected to happbeen

system_error is supposed to be includable without error

- Step-by-step reproduction instructions (by running `brew install` commands)

basically build icu4c or any program that uses std::system_errror
